src/Data/Entity.php

```diff
@@ -21,17 +25,55 @@ public function getId()
         return $this->id;
     }
 
+    public function has($field)
+    {
+        return array_key_exists($field, $this->properties) || in_array($field, $this->properties);
+    }
+
+    public function getDate($field)
+    {
+        return new Carbon($this->{$field});
+    }
+
+    public function getData()
+    {
+        $object = [];
+        foreach ($this->properties as $k=>$v) {
+            if (is_numeric($k)) {
+                $k = $v;
+                $v = null;
+            }
+            $object[$k] = $this->{$k};
+        }
+
+        return $object;
+    }
+
     public function mapData(array $row)
     {
-        $this->data = $row;
+        foreach ($this->properties as $k=>$v) {
```

Comment:
Based on yours and @shadowhand's input I'm going to redo #1, something a little more concise.